Shoko.Server/API/v1/Implementations/ShokoServiceImplementation/ShokoServiceImplementation.cs

```diff
@@ -486,13 +479,30 @@ public CL_Response SaveServerSettings(CL_ServerSettings contractIn)
                 }
             }
 
-            if (contractIn.AniDB_ServerAddress != settings.AniDb.ServerAddress)
+            var serverAddressGroups = Regex.Match(settings.AniDb.HTTPServerUrl, @"(?<=https?://)(?<address>.*):(?<port>\d+)").Groups;
```

Comment:
I would move that to a private readonly compiled regex.

Something like:
```cs
private readonly Regex _urlRegex = new Regex("(?<=https?://)(?<address>.*):(?<port>\d+)", RegexOptions.Compiled);
```

Called with
```cs
serverAddressGroups = _urlRegex.Match(settings.AniDb.HTTPServerUrl).Groups;
```

I'm going from memory, so it might not be quite right